TFT/src/User/API/menu.c

```diff
@@ -819,7 +819,16 @@ void menuDrawTitle(const uint8_t *content)
   GUI_SetBkColor(infoSettings.title_bg_color);
   if (content)
   {
-    GUI_DispLenString(10, start_y, content, LCD_WIDTH - 20, true);
+    if (MENU_IS(menuPrinting) && infoSettings.filename_extension == 0)
+    {
+      hideFileExtension((char *)content);
```

Comment:
you cannot use hideFileExtension/restoreFileExtension on the general function menuDrawTitle. "content" is a general string so it does not provide any extension flag

Reply:
> you cannot use hideFileExtension/restoreFileExtension on the general function menuDrawTitle. "content" is a general string so it does not provide any extension flag.

There's a conditional there. Where does "counter" points when the conditional is true? To what "string"?

> If you want to hide the file extension reported in the title bar (e.g. on Print or PrintingMenu) you can simply make a copy of infoFile.title first and then apply hideFileExtension but not restoreFileExtension. E.g.:
> 
> char copy_of_title[MAX_PATH_LEN];
> 
> strcpy(copy_of_title, infoFile.title); // infoFile.title has the file extension and no extension flag hideFileExtension(copy_of_title); // copy_of_title has no more file extension and no extension flag menuDrawTitle((uint8_t *)copy_of_title);

Thanks, there are more ways to... peal an orange.

> // in any case don't call restoreFileExtension(copy_of_title) or restoreFileExtension(infoFile.title) because it will access/check a missing extension flag

Sure, thx.

> * rollback the changes made in menu.c

Sorry, I won't. There is some bugfix too.

> * remove restoreFileExtension on vfs.h (it will be used only internally by restoreFileIndexExtension).

Nope

> * restore the previous names for hideFileIndexExtension/restoreFileIndexExtension ("Index" doesn't mean anything for me)
> * according to previous bullet, rename hideFileExtension (e.g. hideExtension)

Done.